src/esp/sensor/PinholeCamera.cpp

```diff
@@ -55,15 +55,14 @@ bool PinholeCamera::getObservation(gfx::Simulator& sim, Observation& obs) {
 
   renderTarget().renderEnter();
 
-  // Make sure we have memory
-  if (buffer_ == nullptr) {
-    // TODO: check if our sensor was resized and resize our buffer if needed
-    ObservationSpace space;
-    getObservationSpace(space);
-    buffer_ = core::Buffer::create(space.shape, space.dataType);
-  }
-  obs.buffer = buffer_;
+  drawObservation(sim);
```

Comment:
The `renderTarget().renderEnter();` and `renderTarget().renderExit();` calls can be put into `drawObservation` as they are only needed for drawing.